pkg/daemon/daemon.go

```diff
@@ -1601,9 +1612,99 @@ func PersistNetworkInterfaces(osRoot string) error {
 		return fmt.Errorf("failed to run nmstatectl: %w", err)
 	}
 
+	// If the node is using networking in the initrd (e.g. for Tang pinning), then
+	// it's possible that the network configuration there also relies on specific
+	// network names. To be safe, we'll add kargs to have the names also apply in
+	// the initrd, and then possibly remove them afterwards as part of cleanup.
+
+	if usesInitrdNetworking, err := HostUsesNetworkingInInitrd(); err != nil {
+		return err
+	} else if !usesInitrdNetworking {
+		glog.Info("Not modifying ifname= kargs; host doesn't use initrd networking")
+		return nil // nothing more to do!
+	}
+
+	var args []string
+	if pinning {
+		if ifnames, err = getIfnamesFromLinkFiles(osRoot); err != nil {
+			return err
+		}
+
+		// we just persisted NIC names; we'll need to add kargs
+		for ifname, mac := range ifnames {
+			args = append(args, "--append", "ifname="+ifname+":"+mac)
```

Comment:
I just thought about this but we need to confirm that capitalization of alphabetic characters in the mac address don't matter here.

Reply:
I just normalized to lowercase to be consistent.